framework/test/unit/node/consensus/certificate_generation/commit_pool.spec.ts

```diff
@@ -430,13 +458,60 @@ describe('CommitPool', () => {
 			);
 			expectedCommit = { height, aggregationBits, certificateSignature: aggregateSignature };
 			bftAPI.getBFTParameters.mockReturnValue({
-				validators: [{ address: validatorInfo1.address }, { address: validatorInfo2.address }],
+				validators: [
+					{ address: validatorInfo1.address },
+					{ address: validatorInfo2.address },
+					{ address: validatorInfo3.address },
+				],
 			});
 
 			await expect(
 				commitPool.aggregateSingleCommits(context, singleCommits),
 			).resolves.toStrictEqual(expectedCommit);
 		});
+
+		it('should throw if no bls public key is found for the validator', async () => {
+			const { aggregationBits, signature: aggregateSignature } = createAggSig(
+				validatorKeys,
+				pubKeySignaturePairs,
+			);
+			expectedCommit = { height, aggregationBits, certificateSignature: aggregateSignature };
+			bftAPI.getBFTParameters.mockReturnValue({
+				validators: [{ address: validatorInfo1.address }, { address: validatorInfo2.address }],
+			});
+
+			await expect(commitPool.aggregateSingleCommits(context, singleCommits)).rejects.toThrow(
+				`No bls public key entry found for validatorAddress ${validatorInfo3.address.toString(
+					'hex',
+				)}`,
+			);
+		});
+
+		it.skip('should sort validator keys in lexicographical order', async () => {
+			const unSortedKeys = [validatorKeys[2], validatorKeys[0], validatorKeys[1]];
+			const { aggregationBits, signature: aggregateSignature } = createAggSig(
+				validatorKeys,
+				pubKeySignaturePairs,
+			);
+			const { aggregationBits: aggregationBits2 } = createAggSig(
+				unSortedKeys,
+				pubKeySignaturePairs,
+			);
+			expectedCommit = { height, aggregationBits, certificateSignature: aggregateSignature };
+			bftAPI.getBFTParameters.mockReturnValue({
+				validators: [
+					{ address: validatorInfo1.address },
+					{ address: validatorInfo2.address },
+					{ address: validatorInfo3.address },
+				],
+			});
+
+			expect(aggregationBits).not.toStrictEqual(aggregationBits2);
+
+			await expect(
+				commitPool.aggregateSingleCommits(context, singleCommits),
```

Comment:
As mentioned above, you can spy on `createAggSig` function and ensure it's called with the correct order of the keys. the resulting signature is computed from the single commits, so the signature will be the same regardless of the order of the keys